### ISSUE TITLE
chees_adaptation: float() on a traced step_size makes mass_matrix_estimation unusable under jit

### ISSUE THREAD

**Issue by suchethac** [blackjax-devs/blackjax#1029]
### What happened

`chees_adaptation(...).run` raises `ConcretizationTypeError` under `jax.jit` whenever `mass_matrix_estimation` is set, because `_apply_length_floor` is called with `float(step_size_ma)` on a traced array.

`chees_adaptation.py` (1.6.2, around line 989):

```python
consumed_trajectory_length_ma, floor_clipped_by_cap = _apply_length_floor(
    trajectory_length_ma,
    final_eig_state.lambda_max,
    final_engaged,
    _length_floor,
    max_leapfrog_steps,
    float(step_size_ma),          # <-- step_size_ma = jnp.exp(...), a tracer under jit
)
```

`step_size_ma` is computed a few lines above as `jnp.exp(last_adaptation_state.log_step_size_moving_average)`, so it is a tracer whenever `run` is traced.

The branch is reached only when the trajectory-length floor is engaged, and `enable_length_floor = estimate_mass_matrix and _length_floor` (line ~735) — so it fires exactly when a mass matrix is estimated, and never otherwise. That makes `mass_matrix_estimation="diagonal"` effectively unusable for anyone who jits their warmup.

### Minimal reproducer

Standard 3-D Gaussian, no external dependencies beyond blackjax/jax/optax:

```python
import jax, jax.numpy as jnp, optax, blackjax

def logdensity(x):
    return -0.5 * jnp.sum(x ** 2)

NCHAINS, NDIM = 8, 3
positions = jnp.zeros((NCHAINS, NDIM))

def make(mass_matrix_estimation):
    warmup = blackjax.chees_adaptation(
        logdensity, num_chains=NCHAINS,
        mass_matrix_estimation=mass_matrix_estimation,
    )
    optim = optax.adam(0.1)
    def f(key, pos):
        (state, params), _ = warmup.run(key, pos, 0.1, optim, 40)
        return state.position          # arrays only
    return f

for mm in (None, "diagonal"):
    for jit in (False, True):
        f = make(mm)
        g = jax.jit(f) if jit else f
        try:
            g(jax.random.key(0), positions).block_until_ready()
            print(f"mass_matrix_estimation={mm!r:>12}  jit={jit!s:<5}: OK")
        except Exception as e:
            print(f"mass_matrix_estimation={mm!r:>12}  jit={jit!s:<5}: {type(e).__name__}")
```

Output on blackjax 1.6.2 / jax 0.11.0:

```
mass_matrix_estimation=        None  jit=False: OK
mass_matrix_estimation=        None  jit=True : OK
mass_matrix_estimation=  'diagonal'  jit=False: OK
mass_matrix_estimation=  'diagonal'  jit=True : ConcretizationTypeError
```

Only the `("diagonal", jit=True)` cell fails, which isolates it to the `float()` cast rather than to mass-matrix estimation generally.

### Expected

`run` should be traceable with `mass_matrix_estimation` set, as it is without it.

### Possible fix

`_apply_length_floor`'s `step_size` argument is used to form `cap = max_leapfrog_steps * step_size`, which is then compared against `floor_value` inside a `jnp.logical_and(...)` — so it looks like it could take the traced value directly and drop the `float()`. Its `enable` argument is genuinely static (`if not enable: return ...` early), but `step_size` does not appear to need to be. I have not submitted a patch because I am not sure whether the `float()` was load-bearing for something else in that function; happy to open a PR if that reading is right.

### Environment

- blackjax 1.6.2
- jax / jaxlib 0.11.0
- Python 3.12, Linux x86_64, CPU (also reproduces on CUDA)

### How we hit it

Running ChEES-HMC across a batch of independent Bayesian fits, vmapped and jitted. The failure is not specific to that setup — as above, a bare jitted 3-D Gaussian is enough.

**Comment by junpenglao** [comment]
Thanks for reporting!! Fix on the way.
